Plugins/Mute/Plugin.cs

```diff
@@ -0,0 +1,55 @@
+﻿using SharedLibraryCore;
+using SharedLibraryCore.Interfaces;
+
+namespace Mute;
+
+public class Plugin : IPlugin
+{
+    public Plugin(IMetaServiceV2 metaService)
+    {
+        DataManager = new DataManager(metaService);
+    }
+
+    public string Name => "Mute";
+    public float Version => (float) Utilities.GetVersionAsDouble();
+    public string Author => "Amos";
+
+    public const string MuteKey = "IW4MMute";
+
+    public static DataManager? DataManager;
+
+    public Task OnEventAsync(GameEvent gameEvent, Server server)
+    {
+        if (server.GameName != Server.Game.IW4) return Task.CompletedTask;
+
+        switch (gameEvent.Type)
+        {
+            case GameEvent.EventType.Join:
+                DataManager?.ReadPersistentData(gameEvent.Origin);
```

Comment:
After addressing the previous comment any place you reference data manager you can remove the null coalesce